var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py

```diff
@@ -87,26 +87,29 @@ class EcpDataVisSdk(BundlePackage, CudaPackage, ROCmPackage):
     variant("vtkm", default=False, description="Enable VTK-m")
     variant("zfp", default=False, description="Enable ZFP")
 
+    # Language Options
+    variant("fortran", default=True, sticky=True, description="Enable fortran language features.")
+
     ############################################################
     # Dependencies
     ############################################################
     cuda_arch_variants = ["cuda_arch={0}".format(x) for x in CudaPackage.cuda_arch_values]
     amdgpu_target_variants = ["amdgpu_target={0}".format(x) for x in ROCmPackage.amdgpu_targets]
 
     dav_sdk_depends_on(
-        "adios2+shared+mpi+fortran+python+blosc+sst+ssc+dataman",
+        "adios2+shared+mpi+python+blosc+sst+ssc+dataman",
         when="+adios2",
-        propagate=["hdf5", "sz", "zfp"],
+        propagate=["hdf5", "sz", "zfp", "fortran"],
     )
```

Comment:
I'm not familiar with the dependencies and am a bit confused.  I interpreted the need to add `fortran` as a variant to mean the SDK could be built *with and without* `fortran`.  So I was thinking there would be two `day_sdk_depends_on` here:

```
    dav_sdk_depends_on(
         "adios2+shared+mpi+python+blosc+sst+ssc+dataman",
         when="+adios2",
         propagate=["hdf5", "sz", "zfp"],
     )
    dav_sdk_depends_on(
         "adios2+shared+mpi+fortran+python+blosc+sst+ssc+dataman",
         when="+adios2",
         propagate=["hdf5", "sz", "zfp", "fortran"],
     )
```

But perhaps I didn't read the issue closely enough.

Reply:
So the purpose of the `dav_sdk_depends_on` wrapper is, in part, used to automatically add the positive and negative of the variant propagation without having to write out the boiler plate by hand. Adding `"fortran"` to the list of variants adds 

```python
depends_on("adios2 +fortran", when="+adios2+fortran")
depends_on("adios2 ~fortran", when="+adios2~fortran")
```

This was done because the nature of the SDK requires propagating a  lot of variants and it is far easier to read the intent quickly this way, rather than having to parse out all of the variations on every line.

Reply:
I should've looked more closely at `dav_sdk_depends_on`, especially lines 28-50.